clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/DataRecoveryClient.java

```diff
@@ -45,17 +56,69 @@ public DataRecoveryMonitor getMonitor() {
     return monitor;
   }
 
+  public Map<String, Pair<Boolean, String>> getRepushViability(
+      Set<String> storesList,
+      StoreRepushCommand.Params params) {
+    Map<String, Pair<Boolean, String>> ret = new HashMap<>();
+    String url = params.getUrl();
+    ControllerClient cli = params.getPCtrlCliWithoutCluster();
+    LocalDateTime timestamp = params.getTimestamp();
+    String destFabric = params.getDestFabric();
+    for (String s: storesList) {
+      String clusterName = cli.discoverCluster(s).getCluster();
+      try (ControllerClient parentCtrlCli = buildControllerClient(clusterName, url, params.getSSLFactory())) {
+        StoreHealthAuditResponse storeHealthInfo = parentCtrlCli.listStorePushInfo(s, false);
+        RegionPushDetails regionPushDetails = storeHealthInfo.getRegionPushDetails().get(destFabric);
+        String latestTimestamp = regionPushDetails.getPushStartTimestamp();
+        LocalDateTime latestPushStartTime = LocalDateTime.parse(latestTimestamp, DateTimeFormatter.ISO_LOCAL_DATE_TIME);
+        if (latestPushStartTime.isBefore(timestamp)) {
+          ret.put(s, Pair.of(true, ""));
+        } else {
+          ret.put(s, Pair.of(false, "ongoing repush"));
```

Comment:
if I am not wrong, `listStorePushInfo` only reports the current version info, so to determine if there is an ongoing push, we need a different api call. one example is [here](https://github.com/linkedin/venice/blob/main/clients/venice-admin-tool/src/main/java/com/linkedin/venice/datarecovery/MonitorCommand.java#LL86C20-L86C20)

Reply:
Updated the error handling and added the additional gate based on getFutureVersions